src/algebra/monoid_algebra/degree.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import algebra.monoid_algebra.basic
+
+/-!
+# Lemmas about the `sup` and `inf` of the support of `add_monoid_algebra`
+
+## TODO
+The current plan is to state and prove lemmas about `finset.sup (finsupp.support f) D` with a
+"generic" degree/weight function `D` from the grading Type `A` to a somewhat ordered Type `B`.
+
+Next, the general lemmas get specialized for some yet-to-be-defined `degree`s.
+-/
+
+variables {R A T B ι : Type*}
+
+namespace add_monoid_algebra
+open_locale classical big_operators
+
+/-! ### Results about the `finset.sup` and `finset.inf` of `finsupp.support` -/
+
+section general_results_assuming_semilattice_sup
+variables [semilattice_sup B] [order_bot B] [semilattice_inf T] [order_top T]
+
+section semiring
+variables [semiring R]
+
+section explicit_Ds
+variables (DB : A → B) (DT : A → T) (f g : add_monoid_algebra R A)
```

Comment:
What do you think of calling the functions `degb` and `degt`, and adding a comment that explains the mnemonic `b = bot` and `t = top`?

Reply:
I like this!  Would it be better to call them `bdeg` and `tdeg`, since for `nat/trailing_degree`s the modifier precedes the noun?

Reply:
I think that this is a very good idea!  I used the new names and added a doc-string.